apm-agent-core/src/test/java/co/elastic/apm/agent/report/ApmServerReporterIntegrationTest.java

```diff
@@ -136,17 +164,53 @@ void testSecretToken() throws ExecutionException, InterruptedException {
             exchange.setStatusCode(200).endExchange();
         };
         reporter.report(new Transaction(tracer));
-        reporter.flush().get();
+        reporter.flush();
         assertThat(reporter.getDropped()).isEqualTo(0);
         assertThat(receivedIntakeApiCalls.get()).isEqualTo(1);
     }
 
     @Test
-    void testReportErrorCapture() throws ExecutionException, InterruptedException {
+    void testReportErrorCapture() {
         reporter.report(new ErrorCapture(tracer));
-        reporter.flush().get();
+        reporter.flush();
+        assertThat(reporter.getDropped()).isEqualTo(0);
+        assertThat(receivedIntakeApiCalls.get()).isEqualTo(1);
+    }
+
+    @Test
+    void testTimeout() {
+        doReturn(TimeDuration.of("1ms")).when(reporterConfiguration).getApiRequestTime();
+        doAnswer(invocation -> System.nanoTime()).when(clock).nanoTicks();
```

Comment:
Why is this required? Isn't `System.nanoTime()` the default implementation anyway?

Reply:
I removed the whole clock abstraction as it was only really needed for the soft flush tests.